app/Allocator/Squawk/General/AirfieldPairingSquawkAllocator.php

```diff
@@ -83,9 +80,32 @@ public function fetch(string $callsign): ?SquawkAssignmentInterface
         return AirfieldPairingSquawkAssignment::find($callsign);
     }
 
-    public function canAllocateForCategory(string $category): bool
+    public function assignToCallsign(string $code, string $callsign): ?SquawkAssignmentInterface
     {
-        return $category === SquawkAssignmentCategories::GENERAL;
+        $assignment = null;
+        AirfieldPairingSquawkRange::all()->each(
+            function (AirfieldPairingSquawkRange $range) use ($code, $callsign, &$assignment) {
+                if ($range->squawkInRange($code)) {
+                    // Lock the range to prevent a dupe assignment
+                    AirfieldPairingSquawkRange::lockForUpdate($range->id)->first();
+
+                    // Assign the code, if we can.
+                    if (!AirfieldPairingSquawkAssignment::where('code', $code)->first()) {
+                        $assignment = $this->assignSquawkFromAvailableCodes(
+                            $callsign,
+                            new Collection([$code])
+                        );
+                    }
+
+                    // In any case, if we've found the range with the code, we shouldn't try others.
+                    return false;
```

Comment:
Are there any possible scenarios where a code could be in multiple ranges?

Reply:
In theory, no - they don't overlap. I can't enforce that in the database, but within a particular assignment strategy (CCAMS, ORCAM) they won't occur in more than one range.